priam/src/main/java/com/netflix/priam/cluster/management/Compaction.java

```diff
@@ -0,0 +1,169 @@
+/**
+ * Copyright 2018 Netflix, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.priam.cluster.management;
+
+import com.netflix.priam.IConfiguration;
+import com.netflix.priam.defaultimpl.CassandraOperations;
+import com.netflix.priam.merics.CompactionMeasurement;
+import com.netflix.priam.merics.IMetricPublisher;
+import com.netflix.priam.scheduler.CronTimer;
+import com.netflix.priam.scheduler.TaskTimer;
+import com.netflix.priam.utils.JMXConnectorMgr;
+import org.apache.commons.lang3.StringUtils;
+import org.quartz.CronExpression;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.regex.Pattern;
+
+/**
+ * Utility class to compact the keyspaces/columnfamilies
+ * Created by aagrawal on 1/25/18.
+ */
+@Singleton
+public class Compaction extends IClusterManagement<String> {
+    private static final Logger logger = LoggerFactory.getLogger(Compaction.class);
+    private final IConfiguration config;
+    private static final Pattern columnFamilyFilterPattern = Pattern.compile(".\\..");
+    private final CassandraOperations cassandraOperations;
+
+    @Inject
+    public Compaction(IConfiguration config, IMetricPublisher metricPublisher, CassandraOperations cassandraOperations) {
+        super(config, Task.COMPACTION, metricPublisher, new CompactionMeasurement());
+        this.config = config;
+        this.cassandraOperations = cassandraOperations;
+    }
+
+    public final Map<String, List<String>> updateCompactionCFList(IConfiguration config) throws Exception {
+        final Map<String, List<String>> columnFamilyFilter = new HashMap<>(); //key: keyspace, value: a list of CFs within the keyspace
+
+        if (config.getCompactionCFList() == null || config.getCompactionCFList().isEmpty()) {
```

Comment:
When no filters are provided, can we just call `nt compact` equivalent, so that C* would take care of running parallel compactions as needed and as the #concurrent threads available for it to compact?

Reply:
No, dont pass System keyspaces for compaction. 👍